File: 프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.js
function solution(s) {
    if(s.length % 2) return 0;

    let answer = 0;
    const pair = { "}":"{", "]":"[", ")":"("};

    for (let i = 0; i < s.length; i++) {
        const stack = [];
        const rotS = s.slice(i) + s.slice(0, i);

        for (let j of rotS) {
            if (stack.length && stack[stack.length-1] === pair[j]) stack.pop()
            else stack.push(j)
        }
        if (!stack.length) answer++;
    }

    return answer;
}

File: 프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.py
def solution(s):
    answer = 0
    n = len(s)
    
    for i in range(n):
        stack = []
        for j in range(n):
            cur = s[(i+j) % n]
            if cur == "(" or cur == "[" or cur == "{": stack.append(cur)
            else:
                if not stack: break
        
                if cur == ")" and stack[-1] == "(": stack.pop()
                elif cur == "]" and stack[-1] == "[": stack.pop()
                elif cur == "}" and stack[-1] == "{": stack.pop()
                else: break
        else:
            if not stack: answer += 1
        
    return answer